src/Exporter/DataSource.php

```diff
@@ -33,18 +33,6 @@ public function createIterator(BaseProxyQueryInterface $query, array $fields): \
         $query->getQueryBuilder()->distinct();
         $query->getQueryBuilder()->select($rootAlias);
 
-        $sortBy = $query->getSortBy();
-
-        // AddSelect is needed when exporting the results sorted by a column that is part of ManyToOne relation
-        // @see https://github.com/sonata-project/SonataDoctrineORMAdminBundle/issues/1586
-        if (null !== $sortBy) {
-            $rootAliasSortBy = strstr($sortBy, '.', true);
-
-            if ($rootAliasSortBy !== $rootAlias) {
-                $query->getQueryBuilder()->addSelect($rootAliasSortBy);
-            }
-        }
```

Comment:
Since we remove the fix for https://github.com/sonata-project/SonataDoctrineORMAdminBundle/issues/1586, 
can you try this branch @willemverspyck ?

Reply:
nope, don't try it , test don't even pass. I don't know how to fix both errors. I would need to know on the DataSource if the sort is not a many to one or a one to one so it should be removed.

Reply:
It is working on the list, but not on the export, because on the export we do a distinct and we iterate over the results. On the list we execute the query normally

Reply:
I would say it will require to access the metadata like we did here https://github.com/sonata-project/SonataDoctrineORMAdminBundle/blob/4.x/src/FieldDescription/FieldDescriptionFactory.php

In the `getDoctrineQuery` we're doing
```
$rootEntity = current($queryBuilder->getRootEntities());
        if (false === $rootEntity) {
            throw new \RuntimeException('There are not root entities defined in the query.');
        }

        $identifierFields = $queryBuilder
            ->getEntityManager()
            ->getMetadataFactory()
            ->getMetadataFor($rootEntity)
            ->getIdentifierFieldNames();
```

We might need to play again with 
```
$queryBuilder
            ->getEntityManager()
            ->getMetadataFactory()
            ->getMetadataFor()
```